tests/components/mqtt/test_device_tracker.py

```diff
@@ -429,6 +429,84 @@ async def test_setting_device_tracker_location_via_lat_lon_message(
     assert state.state == STATE_UNKNOWN
 
 
+async def test_setting_device_tracker_location_via_auto_zone_message(
+    hass, mqtt_mock_entry_no_yaml_config, caplog
+):
+    """Test the setting of the latitude and longitude via MQTT."""
+    await mqtt_mock_entry_no_yaml_config()
+    async_fire_mqtt_message(
+        hass,
+        "homeassistant/device_tracker/bla/config",
+        "{ "
+        '"name": "test", '
+        '"state_topic": "test-topic", '
+        '"json_attributes_topic": "attributes-topic" '
+        "}",
+    )
+
+    hass.states.async_set(
+        "zone.school",
+        "zoning",
+        {"latitude": 30.0, "longitude": -100.0, "radius": 100, "friendly_name": "School"},
+    )
+
+    assert await async_setup_component(
+        hass, "zone", {"zone": [{
+            "name": "School",
+            "latitude": 30.0,
+            "longitude": -100.0,
+            "radius": 100
+        }]}
+    )
+
+    await hass.async_block_till_done()
+
+    state = hass.states.get("device_tracker.test")
+    assert state.attributes["source_type"] == "gps"
+
+    assert state.state == STATE_UNKNOWN
+
+    hass.config.latitude = 32.87336
+    hass.config.longitude = -117.22743
+
+    # test auto zone and gps attributes
+    async_fire_mqtt_message(
+        hass,
+        "attributes-topic",
+        '{"latitude":32.87336,"longitude": -117.22743, "gps_accuracy":1.5}',
+    )
+    async_fire_mqtt_message(hass, "test-topic", "auto_zone")
+
+    state = hass.states.get("device_tracker.test")
+    assert state.attributes["latitude"] == 32.87336
+    assert state.attributes["longitude"] == -117.22743
+    assert state.attributes["gps_accuracy"] == 1.5
+    assert state.attributes["source_type"] == "gps"
+    assert state.state == STATE_HOME
+
+    # test manual state override
+    async_fire_mqtt_message(hass, "test-topic", STATE_UNKNOWN)
```

Comment:
This would be the place to publish the special payload

Reply:
Update: I see you do that later. When is the intention of this? I would use some other custom location name, not `STATE_UNKNOWN`

Reply:
This is to test if it correctly sets "back" to an inferred location when switching between unknown/unavailable and auto_zone, should I make this STATE_UNAVAILABLE to make this clear?

Reply:
Okay,
The topic your are sending to is not directly setting a state. It sets the location or `None` to inherrit. You might use a string in stead of `STATE_UNAVAILABLE`, Value payloads are:
- `PAYLOAD_HOME` (`STATE_HOME` is the default)
- `PAYLOAD_NOT_HOME (`STATE_NOT_HOME` is the default)
- `PAYLOAD_AUTO_ZONE` (`DEFAULT_PAYLOAD_AUTO_ZONE` is the default)
- Any location string (if there are no other matches)
`STATE_UNAVAILABLE` is not used as a special payload.
You could e.g. use `"Work"` as payload. (Just assign the string). To show the manual override works.